### DISCUSSION TITLE
event handlers trigger twice

### DISCUSSION THREAD

**Discussion by wslyvh** [wevm/wagmi#1524] [Q&A] [answered]
Event handlers such as watchBlockNumber seem to trigger twice.

I created a simple demo at
https://codesandbox.io/s/vigilant-lamport-m96e8o?file=/src/App.js

**Comment by jxom** [comment, marked as answer]
This is a [side-effect of React's `StrictMode`](https://beta.reactjs.org/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development).

**Comment by wslyvh** [comment]
Oh, my bad. Seems like I've been mixing up some projects/code snippets. 
The issue was that I forgot to unsubscribe

Thanks for the quick response tho!
